analytics/core.go

```diff
@@ -44,18 +47,20 @@ type AmpObject struct {
 	HookExecutionOutcome []hookexecution.StageOutcome
 	SeatNonBid           []openrtb_ext.SeatNonBid
 	RequestWrapper       *openrtb_ext.RequestWrapper
+	ActivityControl      privacy.ActivityControl
 }
 
 // Loggable object of a transaction at /openrtb2/video endpoint
 type VideoObject struct {
-	Status         int
-	Errors         []error
-	Response       *openrtb2.BidResponse
-	VideoRequest   *openrtb_ext.BidRequestVideo
-	VideoResponse  *openrtb_ext.BidResponseVideo
-	StartTime      time.Time
-	SeatNonBid     []openrtb_ext.SeatNonBid
-	RequestWrapper *openrtb_ext.RequestWrapper
+	Status          int
+	Errors          []error
+	Response        *openrtb2.BidResponse
+	VideoRequest    *openrtb_ext.BidRequestVideo
+	VideoResponse   *openrtb_ext.BidResponseVideo
+	StartTime       time.Time
+	SeatNonBid      []openrtb_ext.SeatNonBid
+	RequestWrapper  *openrtb_ext.RequestWrapper
+	ActivityControl privacy.ActivityControl
```

Comment:
It doesn't look right to add `ActivityControl` control as a field to each event. That's not an aspect of the event, but control logic for how the event should be handled. Analytics Adapters have no business accessing this information.

Reply:
Yes, refactored as discussed